### PR TITLE
Lockable immutable config

### PR DESCRIPTION
**Pull request by seba-aln**

### CODE REVIEW

File: src/PubNub/PNConfiguration.php [hunk 1]
```diff
@@ -14,49 +14,52 @@ class PNConfiguration
...
 
     /** @var  string Publish key provided by PubNub */
-    private $publishKey;
+    private ?string $publishKey = null;
 
     /** @var  string Secret key provided by PubNub */
-    private $secretKey;
+    private ?string $secretKey = null;
 
     /** @var  string */
```
Comment (marcin-cebo) [inline src/PubNub/PNConfiguration.php:29]: Do we need this comment now that we have authKey typed?

The same question for other properties.

File: src/PubNub/PNConfiguration.php [hunk 2]
```diff
@@ -432,8 +468,27 @@ public function setUseRandomIV($useRandomIV)
         return $this;
     }
 
-    private function validateNotEmptyString($value)
+    private function isNotEmptyString($value): bool
     {
         return (is_string($value) && strlen(trim($value)) > 0);
     }
+
+    public function disableImmutableCheck(): self
```
Comment (marcin-cebo) [inline src/PubNub/PNConfiguration.php:476]: What is a purpose of this method?

**Review by marcin-cebo — APPROVED** [review]

File: src/PubNub/PNConfiguration.php (hunk 2 above)
Comment (seba-aln) [reply src/PubNub/PNConfiguration.php:482]: so you can disable checking and on your own responsibility have a mutable config

File: src/PubNub/PNConfiguration.php (hunk 1 above)
Comment (seba-aln) [reply src/PubNub/PNConfiguration.php]: I've added descriptions to them from the docs

File: src/PubNub/PNConfiguration.php (hunk 2 above)
Comment (marcin-cebo) [reply src/PubNub/PNConfiguration.php:495]: ok. Thanks.

**Review by marcin-cebo — APPROVED** [review]

**Comment by seba-aln** [comment]
@pubnub-release-bot  release

**Pull request merged**